t/test.pl

```diff
@@ -1733,16 +1733,42 @@ sub warning_like {
 #        _AFTER_ the 'threads' module is loaded.
 sub watchdog ($;$)
 {
+    CORE::state $watchdog;
+    CORE::state $watchdog_thread;
```

Comment:
From the top of test.pl:
```
# Do not rely on features found only in more modern Perls here, as some CPAN
# distributions copy this file and must operate on older Perls. Similarly, keep
# things, simple as this may be run under fairly broken circumstances.
```
I think that means we shouldn't be using state here.

Reply:
Changed to a closurfe